src/frontend/src/flows/recovery/recoveryWizard.ts

```diff
@@ -127,36 +134,125 @@ const addDeviceWarningTemplate = ({
+  identityMetadata,
+  pinIdentityMaterial,
+  nowInMillis,
+}: {
+  credentials: AnchorCredentials;
+  identityMetadata: IdentityMetadata | undefined;
+  pinIdentityMaterial: PinIdentityMaterial | undefined;
+  nowInMillis: number;
+}): DeviceStatus | "no-warning" => {
+  const ONE_WEEK_MILLIS = 7 * 24 * 60 * 60 * 1000;
+  const oneWeekAgoTimestamp = nowInMillis - ONE_WEEK_MILLIS;
+  const hasNotSeenRecoveryPageLastWeek =
+    (identityMetadata?.recoveryPageShownTimestampMillis ?? 0) <
+    oneWeekAgoTimestamp;
+  const showWarningPageEnabled = isNullish(
+    identityMetadata?.doNotShowRecoveryPageRequestTimestampMillis
+  );
+  const totalDevicesCount =
+    credentials.credentials.length + credentials.recovery_credentials.length;
+  if (
+    totalDevicesCount <= 1 &&
+    hasNotSeenRecoveryPageLastWeek &&
+    showWarningPageEnabled
+  ) {
+    if (totalDevicesCount === 0 && !pinIdentityMaterial) {
+      // This should never happen because it means that the user has no devices and no pin.
+      // But we still handle it to avoid a crash assuming there was an error retrieving the pin material.
+      return "pin-only";
+    }
+    return totalDevicesCount === 0 ? "pin-only" : "one-device";
+  }
+  return "no-warning";
+};
+
 // TODO: Add e2e test https://dfinity.atlassian.net/browse/GIX-2600
 export const recoveryWizard = async (
   userNumber: bigint,
   connection: AuthenticatedConnection
 ): Promise<void> => {
   // Here, if the user doesn't have any recovery device, we prompt them to add
   // one.
-  const [recoveries, identityMetadata] = await withLoader(() =>
-    Promise.all([
-      connection.lookupRecovery(userNumber),
-      connection.getIdentityMetadata(),
-    ])
+  const [credentials, identityMetadata, pinIdentityMaterial] = await withLoader(
+    () =>
+      Promise.all([
+        connection.lookupCredentials(userNumber),
+        connection.getIdentityMetadata(),
+        idbRetrievePinIdentityMaterial({
+          userNumber,
+        }),
+      ])
   );
-
-  const ONE_WEEK_MILLIS = 7 * 24 * 60 * 60 * 1000;
   const nowInMillis = Date.now();
-  const oneWeekAgoTimestamp = nowInMillis - ONE_WEEK_MILLIS;
-  const hasNotSeenRecoveryPageLastWeek =
-    (identityMetadata?.recoveryPageShownTimestampMillis ?? 0) <
-    oneWeekAgoTimestamp;
-  if (recoveries.length === 0 && hasNotSeenRecoveryPageLastWeek) {
+
+  const devicesStatus = getDevicesStatus({
+    credentials,
+    identityMetadata,
+    pinIdentityMaterial,
+    nowInMillis,
+  });
+
+  if (devicesStatus !== "no-warning") {
     // `await` here doesn't add any waiting time beacause we already got the metadata earlier.
     await connection.updateIdentityMetadata({
       recoveryPageShownTimestampMillis: nowInMillis,
     });
-    const doAdd = await addPhrase({ intent: "securityReminder" });
-    if (doAdd !== "cancel") {
-      doAdd satisfies "ok";
-
-      await setupRecovery({ userNumber, connection });
+    const userChoice = await addDeviceWarning({
```

Comment:
These warning screens are now inconsistent with the messaging on the manage page when a user only has a pin identity: the manage page strongly suggests setting up a recovery phrase, whereas this screen only nudges with regards to a passkey.

Furthermore, the warning on the management page even persists after having added 2 passkeys. We should reconsider the warning screens on the management page too.